src/components/Calendars/ClientScheduleAppointment.tsx

```diff
@@ -0,0 +1,432 @@
+'use client';
+
+import { Availability } from '@/types/Availability';
+import React, { useState, useEffect, useCallback } from 'react';
+
+type ClientScheduleAppointmentProps = {
+    availabilities: Availability[];
+};
+
+export default function ClientScheduleAppointment({ availabilities }: ClientScheduleAppointmentProps) {
+    const MONTH_NAMES = [
+        'Janeiro',
+        'Fevereiro',
+        'Março',
+        'Abril',
+        'Maio',
+        'Junho',
+        'Julho',
+        'Agosto',
+        'Setembro',
+        'Outubro',
+        'Novembro',
+        'Dezembro',
+    ];
+
+    const DAYS = ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb'];
+
+    const [month, setMonth] = useState(new Date().getMonth());
+    const [day, setDay] = useState(new Date().getDate());
+    const [year, setYear] = useState(new Date().getFullYear());
+    const [no_of_days, setNumDays] = useState<number[]>([]);
+    const [blankdays, setBlankDays] = useState<number[]>([]);
+    const [datepickerValue, setDatepickerValue] = useState('');
+    const [selectedHour, setSelectedHour] = useState<number | null>();
+
+    const [step, setStep] = useState(0);
+    const [availabilitiesMap, setAvailabilities] = useState<Map<string, number[]>>();
+    const [availableHours, setAvailableHours] = useState<number[]>();
+
+    // Calculates the number of days in the given month
+    const getNoOfDays = useCallback(
+        (month: number) => {
+            let daysInMonth = new Date(year, month + 1, 0).getDate();
```

Comment:
why is this month + 1? I don't understand the logic of it :) 

Also, all these `let` can be `const`

Reply:
![image](https://github.com/Nutriway/nutriway-nextjs/assets/35170730/61eff4a5-f259-4d28-8870-72e5e2e5764c)

Reply:
This was the explanation I found when re-used de code

Reply:
okay, instead of doing this by end check https://date-fns.org/v2.30.0/docs/getDaysInMonth